### PR TITLE
add task quick pick smoke tests

### LINKED ISSUE #154277: add task smoke tests

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #154277

### CODE REVIEW

File: test/smoke/src/areas/task/task-quick-pick.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Application, Task, Terminal, TerminalCommandId } from '../../../../automation/';
+
+export function setup() {
+	describe.only('Task Quick Pick', () => {
```
Comment (meganrogge) [inline test/smoke/src/areas/task/task-quick-pick.test.ts:9]:
```suggestion
	describe('Task Quick Pick', () => {
```

File: test/smoke/src/main.ts [hunk 2]
```diff
@@ -400,7 +401,8 @@ describe(`VSCode Smoke Tests (${opts.web ? 'Web' : 'Electron'})`, () => {
 	setupSearchTests(logger);
 	setupNotebookTests(logger);
 	setupLanguagesTests(logger);
-	if (opts.web) { setupTerminalTests(logger); } // Not stable on desktop/remote https://github.com/microsoft/vscode/issues/146811
+	if (opts.web) { setupTerminalTests(logger); } // Not stable on desktop/remote https://github.com/microsoft/vscode/issues/14681
+	setupTaskTests(logger);
```
Comment (meganrogge) [inline test/smoke/src/main.ts:405]: maybe should only run on web?

Comment (meganrogge) [reply test/smoke/src/main.ts:405]: not sure what the above issue means

Comment (Tyriar) [reply test/smoke/src/main.ts:405]: We don't interact with the terminal so it's probably fine. I'm not sure what was causing the remaining problems for the terminal tests but I doubt they apply to tasks.

File: test/smoke/src/areas/task/task-quick-pick.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Application, Task, Terminal, TerminalCommandId } from '../../../../automation/';
+
+export function setup() {
+	describe('Task Quick Pick', () => {
```
Comment (Tyriar) [inline test/smoke/src/areas/task/task-quick-pick.test.ts:9]: Lots of nesting here:

```
  VSCode Smoke Tests (Electron)
    Task
      Task Quick Pick
        run
          hide
            ✔ true
            ✔ false
            ✔ undefined
          icon
            - icon
            - color
            - icon & color
```

Proposal to align closer to what the terminal does:

```
  VSCode Smoke Tests (Electron)
    Task
      runTask command
          hide property - true
          hide property - false
          hide property - undefined
          icon - icon only
          icon - color only
          icon - icon & color
```

File: test/smoke/src/main.ts [hunk 4]
```diff
@@ -400,7 +401,8 @@ describe(`VSCode Smoke Tests (${opts.web ? 'Web' : 'Electron'})`, () => {
 	setupSearchTests(logger);
 	setupNotebookTests(logger);
 	setupLanguagesTests(logger);
-	if (opts.web) { setupTerminalTests(logger); } // Not stable on desktop/remote https://github.com/microsoft/vscode/issues/146811
+	if (opts.web) { setupTerminalTests(logger); } // Not stable on desktop/remote https://github.com/microsoft/vscode/issues/14681
```
Comment (Tyriar) [inline test/smoke/src/main.ts:404]:
```suggestion
	if (opts.web) { setupTerminalTests(logger); } // Not stable on desktop/remote https://github.com/microsoft/vscode/issues/146811
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**